hw1/02-palindrome.js

```diff
@@ -1,3 +1,46 @@
-const elem = document.querySelector('input');
+const elem = document.querySelector("input");
 
-elem.addEventListener('input', handleInput);
+elem.addEventListener("input", handleInput);
+
+//creating a HTML paragraph element
+const paragraph = document.createElement("p");
+
+//appending paragraph as child element of div
+document.getElementById("result").appendChild(paragraph);
+
+paragraph.id = "para";
+
+//reference ---https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event
+//reference ---https://www.geeksforgeeks.org/how-to-check-whether-a-passed-string-is-palindrome-or-not-in-javascript/
+//reference ---https://foolishdeveloper.com/how-to-create-a-palindrome-checker-in-javascript/
+//reference ---https://www.tutorialspoint.com/How-to-change-the-font-color-of-a-text-using-JavaScript
+
+function handleInput(e) {
+  const inp = e.target.value;
+
+  let halfLen = Math.floor(inp.length / 2);
+  let checkPalindrome;
+
+  //check if input is a positive number
+  if (e.target.value > 0) {
+    let j = inp.length - 1;
+    for (let i = 0; i < halfLen; i++) {
```

Comment:
When the input is === 1, the value of halfLen === 0. The palindrome's that are just a single number do not show up because of this. Everything else looks great.